api/graphql/mutations/user.js

```diff
@@ -1,84 +1,194 @@
-import jwt from 'jsonwebtoken'
 import request from 'request'
-import { getPublicKeyFromPem } from '../../../lib/util'
 
-export const createSession = (userId, fetchOne, { req }) => async (root, { email, password }) => {
+// Sign-up Related
+
+export const sendEmailVerification = async (_, { email }) => {
   try {
-    const isLoggedIn = await UserSession.isLoggedIn(req)
-    if (isLoggedIn) {
-      return {
-        me: fetchOne('Me', userId),
-        error: 'already logged-in'
+    let user = await User.find(email, {}, false)
+
+    // Email already validated: 
+    // send failure without error to not reveal account existence
+    if (user?.get('email_validated')) {
+      return { success: false }
+    }
+
+    // User is new or exists without a validated email:
+    // send verification email
+    if (!user) {
+      user = await User.create({ email, active: false })
+    }
+
+    // TODO: Check here if a non-expired UserVerificationCode already exists for this user
```

Comment:
i think its ok to just send a new one.

Reply:
Agreed. And -- The sense in extending it would be to reduce user confusion if they go back and end-up retrieving the old code from email accidentally and then it's already expired. I think it'd be better to just keep extending the existing code's expiry as long as it's not already expired... or maybe even if it is. And I'm happy to do nothing with this. The TODO was not gonna make this pass anyway :)

Reply:
we couldn't extend the expiration of the JWT though, so then we would need to make that one last longer to start with. not sure of all the security implications...

Reply:
Fair. So doing what I was suggesting would result in the code remaining valid in the earlier email but not the token. In the UX this could be fine except currently you'd get an error that the token was "expired" even though you could then enter the code and it'd sail through. I am very happy to do nothing with this for now, but there may be a way through that is still smooth without trying to monkey with the JWT which def doesn't sound like a good idea or even possible :)

Reply:
i just mean we could make the original JWT expiration time longer, but that may not be desired security wise and still would expire eventually